src/components/header/Header.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from 'react';
+import Input from '../input/Input';
+import Button from '../button/Button';
+import '../components.css';
+import { useNavigate } from 'react-router-dom';
+
+interface HeaderState {
+  inputValue: string;
+  updateAppInputValue: (newValue: string) => void;
+  handleStartSearch: () => void;
+  handleStopSearch: () => void;
+}
+
+const Header: React.FC<HeaderState> = (props) => {
```

Comment:
It's better and more convenient to use destructuring for props:
const Header: React.FC<HeaderState> = ({ inputValue, updateAppInputValue, etc }) => {